app/collections/favorites-button.jsx

```diff
@@ -0,0 +1,183 @@
+      .then(([favorites]) => { return (favorites || null); });
+  }
+
+  favoriteSubject(isFavorite) {
+    const favorited = isFavorite;
+    this.setState({ favorited });
+  }
+
+  addSubjectTo(collection) {
+    this.setState({ favorited: true });
+    collection.addLink('subjects', [this.props.subject.id.toString()]);
+  }
+
+  removeSubjectFrom(collection) {
+    this.setState({ favorited: false });
+    collection.removeLink('subjects', [this.props.subject.id.toString()]);
+  }
+
+  getFavoritesName(project) {
+    if (project) {
+      return `Favorites ${project.slug}`;
+    }
+
+    return 'Favorites';
+  }
+
+  logSubjLike(liked) {
+    if (this.context.geordi) {
+      this.context.geordi.logEvent({ type: liked });
+    }
+  }
+
+  createFavorites() {
+    return (
+      new Promise((resolve, reject) => {
+        this.findFavoriteCollection()
+          .catch((err) => { reject(err); })
+          .then((favorites) => {
+            if (favorites) {
+              this.setState({ favorites });
+              resolve(favorites);
+            } else {
+              const display_name = this.getFavoritesName(this.props.project);
+              const project = this.props.project.id;
+              const subjects = [];
+              const favorite = true;
+
+              const links = { subjects };
+              if (project) {
+                links.projects = [project];
+              }
+              const collection = { favorite, display_name, links };
+              apiClient.type('collections')
+                .create(collection)
+                .save()
+                .catch((err) => { reject(err); })
+                .then((newFavorites) => {
+                  this.setState({ favorites: newFavorites });
+                  resolve(newFavorites);
+                });
+            }
+          });
+      })
+    );
+  }
+
+  toggleFavorite() {
+    if (this.props.user) {
+      if (!this.state.favorites) {
+        this.setState({ favorited: true });
+        this.createFavorites()
+          .then((favorites) => { this.addSubjectTo(favorites); });
+        this.logSubjLike('favorite');
+      } else if (this.state.favorited) {
+        this.removeSubjectFrom(this.state.favorites);
+        this.logSubjLike('unfavorite');
+      } else {
+        this.addSubjectTo(this.state.favorites);
+        this.logSubjLike('favorite');
+      }
+    } else {
+      this.promptToSignIn();
+      this.logSubjLike('favorite');
+    }
+  }
+
+  render() {
+    const iconClasses = classnames({
+      'favorited': this.state.favorited,
+      'fa fa-heart': this.state.favorited,
+      'fa fa-heart-o': !this.state.favorited,
+      'fa-fw': true
+    });
+
+    return (
+      <button
+        className={`favorites-button ${this.props.className || ''}`}
+        type="button"
+        title={(this.state.favorited) ? 'Unfavorite' : 'Favorite'}
+        onClick={this.toggleFavorite}
```

Comment:
Add `aria-label={(this.state.favorited) ? 'Unfavorite' : 'Favorite'}` here since the button doesn't have a text label.